packages/hoppscotch-common/src/services/persistence/validation-schemas/index.ts

```diff
@@ -358,7 +393,11 @@ const HoppTestResultSchema = z
           .object({
             additions: z.array(EnvironmentVariablesSchema),
             updations: z.array(
-              EnvironmentVariablesSchema.extend({ previousValue: z.string() })
+              EnvironmentVariablesSchema.and(
+                z.object({
+                  previousValue: z.string(),
+                })
+              )
             ),
```

Comment:
Is it not necessary to narrow this down to the case where `secret` is set to `false` that has a `value` associated?